protostar/protostar_toml/protostar_toml_section.py

```diff
@@ -1,14 +1,21 @@
 from abc import ABC, abstractmethod
 from pathlib import Path
-from typing import Any, Dict, List, Union
+from typing import Any, Dict, Generic, List, TypeVar, Union
 
 from protostar.protostar_toml.protostar_toml_exceptions import (
     InvalidProtostarTOMLException,
 )
 from protostar.utils.protostar_directory import VersionManager, VersionType
 
+T = TypeVar("T")
```

Comment:
```suggestion
T = TypeVar("T", bound="ProtostarTOMLSection")
```
Wouldn't be stricter? not sure not an expert in `TypeVar`